src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,649 @@
+                @Nullable final String badArgument)
+        {
+            super();
+            this.tabCompletions = new ArrayList<>(tabCompletions);
+            this.executionTreeNode = executionTreeNode;
+            this.executionActionArgumentList = new ArrayList<>(executionActionArgumentList);
+            this.actionMenuState = actionMenuState;
+            this.badArgument = badArgument;
+        }
+
+        public List<String> getTabCompletions()
+        {
+            return Collections.unmodifiableList(tabCompletions);
+        }
+
+        public TreeNode<IMenu> getExecutionTreeNode()
+        {
+            return executionTreeNode;
+        }
+
+        public List<ActionArgument> getExecutionActionArgumentList()
+        {
+            return Collections.unmodifiableList(executionActionArgumentList);
+        }
+
+        public String getBadArgument()
+        {
+            return badArgument;
+        }
+
+        @Nullable
+        public ActionMenuState getActionMenuState()
+        {
+            return actionMenuState;
+        }
+    }
+
+    /**
+     * The level required to execute /mc commands.
+     * private static final int OP_PERMISSION_LEVEL = 3;
+     */
+
+    @NotNull
+    private TreeNode<IMenu> getRoot()
+    {
+        if (null == root)
+        {
+            root = buildMenu(getRootNavigationMenuType());
+        }
+        return root;
+    }
+
+    /**
+     * Removed so we can control this at the config
+     * Only allow OP's execute the commands.
+     *
+     * @return the int permission level (3 for OP).
+     * public int getRequiredPermissionLevel(){return OP_PERMISSION_LEVEL;}
+     */
+    @NotNull
+    @Override
+    public List<String> getAliases()
+    {
+        return Arrays.asList("mc", "col", "mcol", "mcolonies", "minecol", "minecolonies");
+    }
+
+    private TreeNode<IMenu> buildMenu(@NotNull final IMenuType menuType)
+    {
+        return buildMenu(menuType, new HashSet<>());
+    }
+
+    private TreeNode<IMenu> buildMenu(@NotNull final IMenuType menuType, @NotNull final Set<IMenuType> menuTypesSoFar)
+    {
+        // Prevent recursion
+        if (menuTypesSoFar.contains(menuType))
+        {
+            return null;
+        }
+        else
+        {
+            menuTypesSoFar.add(menuType);
+        }
+
+        final IMenu menu = menuType.getMenu();
+        final TreeNode<IMenu> treeNode = new TreeNode<>(menu);
+
+        if (menuType.isNavigationMenu())
+        {
+            final NavigationMenu navigationMenu = (NavigationMenu) menu;
+            for (final IMenuType childMenuType : navigationMenu.getChildrenMenuList())
+            {
+                final TreeNode<IMenu> childTreeNode = buildMenu(childMenuType, menuTypesSoFar);
+                treeNode.addChild(childTreeNode);
+            }
+        }
+
+        return treeNode;
+    }
+
+    @NotNull
+    @Override
+    public String getName()
+    {
+        return getRoot().getData().getMenuItemName();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "getRoot" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)